[WIP] Docs - Support for the .localhost Top-Level Domain (#35989)
dotnet/AspNetCore.Docs: aspnetcore/test/localhost-tld.md

---
author: tdykstra
description: Learn how using localhost as top-level domain can make testing easier in some scenarios.
monikerRange: '>= aspnetcore-10.0'
ms.author: tdykstra
ms.custom: 

> ms.date: 3/24/2025

Reviewer: ```suggestion
ms.date: 08/26/2025
```

title: Support for the .localhost top-level domain
uid: test/localhost-tld
---

# Support for the .localhost top-level domain

The `.localhost` top-level domain (TLD) is defined in [RFC2606](https://www.rfc-editor.org/rfc/rfc2606) and [RFC6761](https://www.rfc-editor.org/rfc/rfc6761) as being reserved for testing purposes and available for users to use locally as they would any other domain name. This means using a name like `myapp.localhost` locally that resolves to the IP loopback address is allowed and expected according to these RFCs. 

Modern evergreen browsers already automatically resolve any `*.localhost` name to the IP loopback address (`127.0.0.1`/`::1`), effectively making them an alias for any service already being hosted at `localhost` on the local machine. Any service responding to `http://localhost:6789` will also respond to `http://anything-here.localhost:6789`, assuming no further specific hostname verification or enforcement is being performed by the service.

> ASP.NET Core in .NET 10 has been updated to better support the `.localhost` TLD, making it easy to use when creating and running ASP.NET Core applications in your local development environment. Having different apps running locally be resolvable via different names allows for better separation of some domain-name-associated website assets (for example, cookies) and makes it easier to identify which app you're browsing via the name displayed in the browser address bar.

Reviewer: ```suggestion
Having different apps running locally be resolvable via different names allows for better separation of some domain-name-associated website assets (for example, cookies) and makes it easier to identify which app you're browsing via the name displayed in the browser address bar. Support for  ASP.NET Core in .NET 10 was introduced in ASP.NET Core in .NET 10. 
```
